src/main/java/life/qbic/subscriptions/encryption/Encrypter.java

```diff
@@ -0,0 +1,26 @@
+package life.qbic.subscriptions.encryption;
+
+import java.util.function.Function;
+
+/**
+ * Encrypts a message of type {@link I} to a message of type {@link O}
+ *
+ * @param <I> input type
+ * @param <O> encoded output type
+ */
+@FunctionalInterface
+public interface Encrypter<I, O> extends Function<I, O> {
+
+  /**
+   * Encrypts a message of type {@link I} to a message of type {@link O}
+   *
+   * @param input message to be encrypted
+   * @return encrypted message
+   */
+  O encrypt(I input);
```

Comment:
shouldn't this function be private? So we then have a functional interface where only the apply function is exposed.

Reply:
@KochTobi same here

Reply:
same answer 😄 
It is nicer to have something like the following:
```
Decrypter<String, String> myRandomVarName = it -> it.substring(0, 2);
myRandomVarName.decrypt("Test") // "Te"
```
so when you later use this interface you have a more explicit call.